net/ipv6/tcp_ipv6.c

```diff
@@ -1384,6 +1384,7 @@ static struct sock *tcp_v6_syn_recv_sock(const struct sock *sk, struct sk_buff *
 	 */
 	if (tempesta_new_clntsk(newsk, skb)) {
 		tcp_v6_send_reset(newsk, skb);
+		tempesta_close_clntsk(newsk);
```

Comment:
You calling `tempesta_close_clntsk() / tfw_classify_conn_close()` in case of failing `tempesta_new_clntsk()`. However `tempesta_new_clntsk()` can fail before  `sk_security` being initialized, if there is no memory in TDB reserved for clients. Therefore `tfw_classify_conn_close()` should handles this case gracefully, without crashes.

`BUG_ON(!ra);` looks unnecessary in this case, but need additional check for `sk_security`.